stories/SideNavigation.stories.js

```diff
@@ -0,0 +1,324 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import { withKnobs, boolean, text, select } from '@storybook/addon-knobs'
+
+import {
+  SideNavigation,
+  Flexy,
+  Heading,
+  Icon,
+  Button,
+  AutoDropdown,
+} from '../src/index.js'
+import Dropdown from '../src/components/Dropdown/DropdownV2'
+import { SideNavigationHeaderUI } from '../src/components/SideNavigation/SideNavigation.css'
+import { ItemSpec } from './DropdownV2.stories'
+
+import styled from '../src/components/styled'
+import { render } from 'enzyme'
+
+const renderSidebarFolders = () => {
+  return (
+    <SideNavigation.Section title="Folders">
+      <SideNavigation.Item>Folder 1</SideNavigation.Item>
+      <SideNavigation.Item>Folder 2</SideNavigation.Item>
+      <SideNavigation.Item>Folder 3</SideNavigation.Item>
+      <SideNavigation.Item>Folder 4</SideNavigation.Item>
+    </SideNavigation.Section>
+  )
+}
+
+const footerItems = ItemSpec.generate(3)
+
+const renderSidebarFooter = isFloating => {
+  return (
+    <SideNavigation.Footer>
+      <Dropdown
+        items={footerItems}
+        renderTrigger={
+          <SideNavigation.Button
+            icon={
+              <Icon
+                name="workflow"
+                offsetLeft={false}
+                withCaret={!isFloating}
+              />
+            }
+          >
+            Edit Mailbox
+          </SideNavigation.Button>
+        }
+      />
+      <SideNavigation.Button icon={<Icon name="image-add" />}>
+        New conversation
+      </SideNavigation.Button>
+    </SideNavigation.Footer>
+  )
+}
+
+export const SideNavigationFloatingUI = styled(SideNavigation)`
+  position: absolute;
+  left: 8px;
+  top: 50px;
+  height: auto;
+`
+
+class SidebarWrapper extends React.PureComponent {
+  render() {
+    return (
+      <Flexy style={{ height: '90vh' }} gap="xl" align="top">
+        <Flexy.Item
+          style={{
+            height: '100%',
+          }}
+        >
+          {this.props.children}
+        </Flexy.Item>
+        <Flexy.Block>
+          <Heading>CONTENT</Heading>
+        </Flexy.Block>
+      </Flexy>
+    )
+  }
+}
+
+class SidebarCollapsed extends React.PureComponent {
+  state = {
+    isCollapsed: true,
+  }
+
+  handleMouseOver = () => {
+    this.setState({ isCollapsed: false })
+  }
+  handleMouseOut = () => {
+    this.setState({ isCollapsed: true })
+  }
+
+  render() {
+    const { badge, headerLabel } = this.props
+
+    return (
+      <SideNavigation
+        collapsed={this.state.isCollapsed}
+        onMouseEnter={this.handleMouseOver}
+        onMouseLeave={this.handleMouseOut}
+      >
+        <SideNavigation.Header label={headerLabel} badge={badge} />
+        <SidebarDefaultItems />
+        {renderSidebarFolders()}
+        {renderSidebarFooter()}
+      </SideNavigation>
+    )
+  }
+}
+
+class SidebarFloatingMenu extends React.PureComponent {
```

Comment:
On the floating feature.

It feels a little too manual to set it up (you have to position it yourself, remember to set up `position: relative` on the parent...) maybe it should be handled by the component itself?

a quick snippet:

```jsx
<SideNavigation
  floating
  floatingTriggerContent="🍔"
  isOpen={this.state.isOpen}
>
</SideNavigationFloatingUI>
```

Just a thought, there might be other things I'm not taking into consideration!

Reply:
made a comment about that here: https://github.com/helpscout/hsds-react/pull/494#issuecomment-474898161